### PR TITLE
Changed request structure for easier adding new endpoints.

### PR DESCRIPTION
**Pull request by troyanryller**

### CODE REVIEW

File: lib/bloom_remit_client.rb [hunk 1]
```diff
@@ -1,29 +1,48 @@
+# frozen_string_literal: true
```
Comment (ramontayag) [inline lib/bloom_remit_client.rb]: What does this mean?

File: lib/bloom_remit_client.rb [hunk 2]
```diff
@@ -1,29 +1,48 @@
...
-require "bloom_remit_client/requests/billers_request"
-require "bloom_remit_client/requests/create_sender_request"
-require "bloom_remit_client/requests/create_payment_request"
-require "bloom_remit_client/responses/base_response"
-require "bloom_remit_client/responses/billers_response"
-require "bloom_remit_client/responses/create_sender_response"
-require "bloom_remit_client/responses/create_payment_response"
+require "bloom_remit_client/requests/credit/list"
+require "bloom_remit_client/requests/credit/history"
+# require "bloom_remit_client/requests/billers_request"
```
Comment (ramontayag) [inline lib/bloom_remit_client.rb]: Did you plan to disable these?

File: lib/bloom_remit_client.rb [hunk 3]
```diff
@@ -1,29 +1,48 @@
...
+  PRODUCTION = 'api.bloom.solutions'
+  STAGING = 'bloomremit-staging.herokuapp.com'
 
-  def self.new(*args)
-    client = Client.new(*args)
-    raise ArgumentError, client.errors.full_messages if client.invalid?
-    client
-  end
+  class << self
+    attr_accessor :sandbox
```
Comment (ramontayag) [inline lib/bloom_remit_client.rb]: I was thinking of controlling this completely via passing in values outside of this gem, but I could be convinced other wise... :)

File: lib/bloom_remit_client/requests.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+module BloomRemitClient
+  module Requests
+    PATH_PARAM_REGEX = /\:[^\/|\?|\#|$]*/
+    GET = 'get'
+    POST = 'post'
+    PUT = 'put'
+    DELETE = 'delete'
+    REQUEST_TYPES = [GET, POST, PUT, DELETE].freeze
+    CONTENT_TYPE_JSON_UTF8 = 'application/json;charset=UTF-8'
+    CONTENT_TYPE_JSON = 'application/json'
```
Comment (ramontayag) [inline lib/bloom_remit_client/requests.rb]: 👍

File: lib/bloom_remit_client/requests/url_builder.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      def call
+        builder.build(to_h).to_s
+      end
+
+      def host
+        return super unless subdomain.present?
+        "#{subdomain}.#{super}"
+      end
+
+      def path
```
Comment (ramontayag) [inline lib/bloom_remit_client/requests/url_builder.rb]: Why move from lazily evaluated Virtus defaults to overriding the reader?

File: lib/bloom_remit_client.rb (hunk 1 above)
Comment (troyanryller) [inline lib/bloom_remit_client.rb]: @ramontayag this makes string literals immutable. Strings immutability will be default in ruby 3.0 
https://wyeworks.com/blog/2015/12/1/immutable-strings-in-ruby-2-dot-3

File: lib/bloom_remit_client.rb (hunk 2 above)
Comment (troyanryller) [inline lib/bloom_remit_client.rb]: according structure changes request files should be like: `"bloom_remit_client/requests/biller/list"`
For now, only implemented `credit` endpoints as example

File: lib/bloom_remit_client.rb (hunk 3 above)
Comment (troyanryller) [inline lib/bloom_remit_client.rb]: @ramontayag actually this allow to set `sandbox true/false` or just specify `host`. 
Ex.: 
`BloomRemitClient.sandbox = true` => will use  'bloomremit-staging.herokuapp.com' as host.
`BloomRemitClient.host = 'example.com'` => will use 'example.com' and ignores `sandbox` variable.

File: lib/bloom_remit_client/requests/url_builder.rb (hunk 5 above)
Comment (troyanryller) [inline lib/bloom_remit_client/requests/url_builder.rb]: @ramontayag what do u mean? Basically, default value for path should be empty line... This override is just normalizer which adds `/` to path string beginning if it not present(`path/without/slash` => `/path/without/slash`) But looks like better to override setter:
```ruby
def path=(value)
  @path = normalize_path(value) 
end
```

File: lib/bloom_remit_client.rb (hunk 2 above)
Comment (ramontayag) [inline lib/bloom_remit_client.rb]: I see. Ok because we're using this now, so I'll need to work on this.

Comment (troyanryller) [inline lib/bloom_remit_client.rb]: @ramontayag can add it in new PR, have it implemented already in `feature/changed-structure-and-add-more-endpoints` branch

File: lib/bloom_remit_client/requests/url_builder.rb (hunk 5 above)
Comment (ramontayag) [inline lib/bloom_remit_client/requests/url_builder.rb]: I found it much easier to test when I could set the attributes directly. Using Virtus' lazy execution allowed for that. Looks like you addressed it already though.

File: lib/bloom_remit_client.rb (hunk 1 above)
Comment (ramontayag) [inline lib/bloom_remit_client.rb]: Cool. Thank you - I didn't know this.

File: lib/bloom_remit_client.rb (hunk 2 above)
Comment (ramontayag) [inline lib/bloom_remit_client.rb]: Ok this is what I was concerned about. Thanks for clarifying.

**Pull request merged**